.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,142 @@
+name: build-and-test
+
+on: [push, pull_request]
+
+concurrency:
+  group: build-and-test-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  PYTHON_VERSION: 3.8.5
+  PIP_VERSION: 20.2.4
+  REQUIREMENTS_PATH: "internal/buildscripts/packaging/tests/requirements.txt"
+  RESULT_PATH: "~/testresults"
+  GO_VERSION: 1.17.2
+
+jobs:
+  setup-environment:
+    name: setup-environment
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Caching dependency
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+
+      - name: Installing dependency
+        run: |
+          make install-tools
+      
+      - name: Zip artifact for deployment
```

Comment:
Do we need to do this if we're already caching?  The subsequent jobs should be able to use the cache.

Reply:
Made changes to use Caching dependency in all subsequent jobs